### PR TITLE
test(front50): Add unit tests on front50 config generation

### PR DESCRIPTION
**Pull request by ezimanyi**
* test(front50): Add unit tests on front50 config generation

* test(front50): Add end-to-end test of YAML generation for front50

### CODE REVIEW

File: pkg/parse_hal/hal_to_clouddriver_test.go [hunk 1]
```diff
@@ -148,7 +148,7 @@ func TestHalToClouddriver(t *testing.T) {
 		t.Run(tt.n, func(t *testing.T) {
 			gotC := parse_hal.HalToClouddriver(tt.h)
 			if !reflect.DeepEqual(gotC, tt.wantC) {
-				t.Errorf("Expected hal config to generate %v, got %v", tt.wantC, gotC)
+				t.Errorf("%s: Expected hal config to generate %v, got %v", tt.n, tt.wantC, gotC)
```
Comment (maggieneterval) [inline pkg/parse_hal/hal_to_clouddriver_test.go]: I thought I remembered the error output including the name of the test right above each error (which is why you have to pass some name to `t.Run`, but if this makes it even more readable then I'm on board!

**Review by maggieneterval — APPROVED** [review]
❗ 🔙 5️⃣ 0️⃣ 📜 ➡️ 📜 💯

File: pkg/parse_hal/hal_to_clouddriver_test.go (hunk 1 above)
Comment (ezimanyi) [reply pkg/parse_hal/hal_to_clouddriver_test.go]: Great point! I somehow completely missed that in the output, but now that I see the name is there I agree it makes more sense to leave it out of the actual error.  Thanks for pointing that out, I'm still getting used to what this test output looks like!

File: pkg/parse_hal/hal_to_clouddriver_test.go [hunk 2]
```diff
@@ -148,7 +148,7 @@ func TestHalToClouddriver(t *testing.T) {
 		t.Run(tt.n, func(t *testing.T) {
 			gotC := parse_hal.HalToClouddriver(tt.h)
```
[the remark this answers is not in the document]
Comment (ezimanyi) [reply pkg/parse_hal/hal_to_clouddriver_test.go]: Yup, good point, made the change.

**Pull request merged**